### PR TITLE
DIOS-6770 & DIOS-6772: App & SDK Version Numbers

### PR DESCRIPTION
**Pull request by sheiladoherty-dolby**
Added version numbers to Interactive Player & tvOS Viewer
- SDK version has to be a hardcoded value that will have to be updated when the SDK is updated. I have been unable to find a dynamic way of doing this.
- For tvOS, I added the versions to the landing screen as the settings it within the the player itself, so I didn't fee like that placement was as helpful. I'm happy to put it elsewhere.

iOS:
Simulator Screenshot - iPhone 15 Pro - 2024-09-27 at 10 16 40

tvOS:
Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-09-27 at 10 15 58

### CODE REVIEW

File: interactive-player/Interactive Viewer/Constants.swift [hunk 1]
```diff
@@ -6,4 +6,5 @@ struct Constants {
     // Demo stream name and account ID
     static let streamName = "multiview"
     static let accountID = "k9Mwad"
+    static let sdkVersion = "2.0.0"
```
Comment (aravind-raveendran) [inline interactive-player/Interactive Viewer/Constants.swift:9]: SDK has an API to get the version number; `MCLogger.getVersion()`

File: rts-viewer-tvos/RTSViewer/StreamDetailInputView/StreamDetailInputViewModel.swift [hunk 2]
```diff
@@ -19,9 +19,19 @@ final class StreamDetailInputViewModel: ObservableObject {
     }
     @Published private(set) var hasSavedStreams: Bool = false
 
+    @Published private(set) var appVersion: String = ""
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/StreamDetailInputView/StreamDetailInputViewModel.swift:22]: if the property is not changing afterwards it does not need to be @Published ?

**Review by aravind-raveendran — COMMENTED** [review]
LGTM

File: interactive-player/Interactive Viewer/Views/Settings/SettingsViewModel.swift [hunk 3]
```diff
@@ -54,6 +57,13 @@ final class SettingsViewModel: ObservableObject {
         streamSortOrderSelectedLabelKey = "place.holder"
         audioSelectedLabelKey = "place.holder"
 
+        if let version = Bundle.main.releaseVersionNumber,
+           let build = Bundle.main.buildVersionNumber {
+            appVersion = "App Version \(version).\(build)"
```
Comment (aravind-raveendran) [inline interactive-player/Interactive Viewer/Views/Settings/SettingsViewModel.swift:62]: In the appstore build `MARKETING_VERSION` becomes of format (1.x.x) and build number is taken from the CI runner number. So if that be formatted as 1.x.x.build_number. 
May be a good formatting is to show similar to that of TestFlight ?  _**1.x.x (build_number)**_

Comment (sheiladoherty-dolby) [reply interactive-player/Interactive Viewer/Views/Settings/SettingsViewModel.swift:62]: @aravind-raveendran How do I get the build number from the CI for the app?

Comment (aravind-raveendran) [reply interactive-player/Interactive Viewer/Views/Settings/SettingsViewModel.swift:62]: It's written to the CFBundleVersion in info.plist. You are already reading the build number right. It's just that the version will be of format 1.x.x . So my suggestion was to format it as "\(version) (\(build))".

Comment (sheiladoherty-dolby) [reply interactive-player/Interactive Viewer/Views/Settings/SettingsViewModel.swift]: Oh, ok, thanks for clarifying. Done.

**Pull request merged**